homeassistant/components/light/tplink.py

```diff
@@ -7,40 +7,34 @@
 import logging
 import time
 
-import voluptuous as vol
-
-from homeassistant.const import (CONF_HOST, CONF_NAME)
 from homeassistant.components.light import (
     Light, ATTR_BRIGHTNESS, ATTR_COLOR_TEMP, ATTR_HS_COLOR, SUPPORT_BRIGHTNESS,
-    SUPPORT_COLOR_TEMP, SUPPORT_COLOR, PLATFORM_SCHEMA)
-import homeassistant.helpers.config_validation as cv
+    SUPPORT_COLOR_TEMP, SUPPORT_COLOR)
 from homeassistant.util.color import \
     color_temperature_mired_to_kelvin as mired_to_kelvin
 from homeassistant.util.color import (
     color_temperature_kelvin_to_mired as kelvin_to_mired)
+import homeassistant.helpers.device_registry as dr
+from homeassistant.components.tplink import DOMAIN as TPLINK_DOMAIN
+
+DEPENDENCIES = ['tplink']
 
-REQUIREMENTS = ['pyHS100==0.3.3']
+PARALLEL_UPDATES = 0
 
 _LOGGER = logging.getLogger(__name__)
 
 ATTR_CURRENT_POWER_W = 'current_power_w'
 ATTR_DAILY_ENERGY_KWH = 'daily_energy_kwh'
 ATTR_MONTHLY_ENERGY_KWH = 'monthly_energy_kwh'
 
-DEFAULT_NAME = 'TP-Link Light'
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_HOST): cv.string,
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
-})
 
+async def async_setup_entry(hass, config_entry, async_add_devices):
```

Comment:
We probably want to keep an empty `async_setup_platform` method around so that people that have it configured don't get exceptions.

Reply:
Changed it to display a warning, but should it still raise a PlatformNotReady exception to make the misconfiguration more visible?

Reply:
`PlatformNotReady` should only be raised when the platform is currently not ready, but might be in the future.

Reply:
Ah, indeed. Or another exception? Or is it better just to be left as it is now?